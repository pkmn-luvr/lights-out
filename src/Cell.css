.Cell {
  height: 60px;
  width: 60px;
  background-color: #b2a8b4;
  border-radius: 50%;
  cursor: pointer;
  display: inline-block;
  margin: 4px;
  transition: background-color 0.3s ease;
}

.Cell-lit {
  background-color: #faeef3; 
  box-shadow: 0 0 15px #ff00bf; 
}

@keyframes flip {
  from {
    transform: scale(1);
  }
  to {
    transform: scale(1.1);
  }
}

.Cell:hover {
  animation-name: flip;
  animation-duration: 0.2s;
  animation-fill-mode: forwards;
}