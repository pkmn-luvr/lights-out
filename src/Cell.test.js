import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Cell from './Cell';


// Smoke test
describe('Cell', () => {
  test('renders without crashing', () => {
    const defaultProps = {
      isLit: true,
      flipCellsAroundMe: () => {} 
    };
    render(<Cell {...defaultProps} />);
  });
});

//Snapshot test
describe('Cell', () => {
    it('renders correctly', () => {
      const defaultProps = {
        isLit: true,
        flipCellsAroundMe: () => {}
      };
      const { asFragment } = render(<Cell {...defaultProps} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
  