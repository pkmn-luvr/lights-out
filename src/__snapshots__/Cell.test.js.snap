// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell renders correctly 1`] = `<DocumentFragment />`;
